packages/salesforcedx-vscode-apex/src/languageUtils/languageClientUtils.ts

```diff
@@ -104,3 +119,137 @@ export const getExceptionBreakpointInfo = async (): Promise<{}> => {
+      const releaseFolders = entries
+        .filter(([name, type]) => type === vscode.FileType.Directory && /^\d{3}$/.test(name))
+        .map(([name]) => name);
+      for (const folder of releaseFolders) {
+        const folderUri = vscode.Uri.joinPath(toolsUri, folder);
+        await vscode.workspace.fs.delete(folderUri, { recursive: true, useTrash: true });
+      }
+    } catch (error) {
+      console.log('Error, failed to delete folder:' + error.message);
+    }
+  }
+};
+
+export const createLanguageClient = async (
+  extensionContext: vscode.ExtensionContext,
+  languageServerStatusBarItem: ApexLSPStatusBarItem
+): Promise<void> => {
+  const telemetryService = await getTelemetryService();
+  // Initialize Apex language server
+  try {
+    const langClientHRStart = process.hrtime();
+    languageClientUtils.setClientInstance(await languageServer.createLanguageServer(extensionContext));
+
+    const languageClient = languageClientUtils.getClientInstance();
+
+    if (languageClient) {
+      languageClient.errorHandler?.addListener('error', (message: string) => {
+        languageServerStatusBarItem.error(message);
+      });
+      languageClient.errorHandler?.addListener('restarting', (count: number) => {
+        languageServerStatusBarItem.error(
+          nls.localize('apex_language_server_quit_and_restarting').replace('$N', `${count}`)
+        );
+      });
+      languageClient.errorHandler?.addListener('startFailed', () => {
+        languageServerStatusBarItem.error(nls.localize('apex_language_server_failed_activate'));
+      });
+
+      // TODO: the client should not be undefined. We should refactor the code to
+      // so there is no question as to whether the client is defined or not.
+      await languageClient.start();
+      // Client is running
+      const startTime = telemetryService.getEndHRTime(langClientHRStart); // Record the end time
+      telemetryService.sendEventData('apexLSPStartup', undefined, {
+        activationTime: startTime
+      });
+      await indexerDoneHandler(retrieveEnableSyncInitJobs(), languageClient, languageServerStatusBarItem);
+      extensionContext.subscriptions.push(languageClientUtils.getClientInstance()!);
+    } else {
+      languageClientUtils.setStatus(
+        ClientStatus.Error,
+        `${nls.localize('apex_language_server_failed_activate')} - ${nls.localize('unknown')}`
+      );
+      languageServerStatusBarItem.error(
+        `${nls.localize('apex_language_server_failed_activate')} - ${nls.localize('unknown')}`
+      );
+    }
+  } catch (e) {
+    let errorMessage = '';
+    if (typeof e === 'string') {
+      errorMessage = e;
+    } else if (e instanceof Error) {
+      errorMessage = e.message ?? nls.localize('unknown_error');
+    }
+    if (errorMessage.includes(nls.localize('wrong_java_version_text', SET_JAVA_DOC_LINK))) {
+      errorMessage = nls.localize('wrong_java_version_short');
+    }
```

Comment:
I think that's big enough to be its own function.  `createLanguageClient` is a bit long

getErrorMessage (e:unknown):string